file/validate.go

```diff
@@ -29,5 +31,70 @@ func validate(content []byte) error {
 		err := fmt.Errorf(desc.String())
 		errs.Errors = append(errs.Errors, err)
 	}
+
+	var kongdefaults *KongDefaults
+	if err := json.Unmarshal(content, kongdefaults); err != nil {
+		return fmt.Errorf("unmarshaling file into KongDefaults: %w", err)
+	}
+	if kongdefaults != nil {
+		if err := CheckDefaults(*kongdefaults); err != nil {
+			return fmt.Errorf("default values are not allowed for these fields: %w", err)
+		}
+	}
 	return errs
 }
+
+func Check(item string, t *string) string {
+	if t != nil && len(*t) > 0 {
+		return item
+	}
+	return ""
+}
+
+func CheckDefaults(kd KongDefaults) error {
+	var invalid []string
+	if ret := Check("Service.ID", kd.Service.ID); len(ret) > 0 {
+		invalid = append(invalid, "Service.ID")
+	}
+
+	if ret := Check("Service.Host", kd.Service.Host); len(ret) > 0 {
```

Comment:
kd.Service is a pointer. What happens if it is a `nil`?

Reply:
checked 67aa05d6a0a1841031b4910928070b31bfaf9be8